### PR TITLE
Sarahday/sc 154166/update flutter sdk hello app readme and comments

### PR DESCRIPTION
**Pull request by scribblingfox**

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -24,21 +27,37 @@ class _HelloAppState extends State<HelloApp> {
...
+      // soon after you run the demo.
+      final LDUser user = LDUserBuilder('example-user-key')
+      .name('Sandy');
+      .build();
       LDClient.start(config, user).whenComplete(() {
           updateFlagEvaluation();
           LDClient.registerFeatureFlagListener(FEATURE_FLAG_KEY, updateFlagEvaluation);
-      });
+      }
+      setState(() { message = 'SDK successfully initialized!'});
```
Comment (scribblingfox) [inline lib/main.dart:40]: Not sure this is the right place for this--I couldn't tell if the SDK initialization happened during the user identification step, or before.

File: lib/main.dart [hunk 2]
```diff
@@ -24,21 +27,37 @@ class _HelloAppState extends State<HelloApp> {
...
+      )
     } else {
-      setState(() { message = 'MOBILE_SDK_KEY not set'; });
+      setState(() { message = 'Please edit lib/main.dart to set MOBILE_SDK_KEY to your LaunchDarkly SDK key first'; });
+      setState(() { message = 'SDK failed to initialize'});
```
Comment (scribblingfox) [inline lib/main.dart:44]: Similar here--couldn't tell if the absent SDK key would cause an initialization failure or not.

File: lib/main.dart [hunk 3]
```diff
@@ -24,21 +27,37 @@ class _HelloAppState extends State<HelloApp> {
...
+    setState(() { message = 'Feature flag "$FEATURE_FLAG_KEY" is ' + result.toString(); ' for this user'});
   }
 
+  // Here we ensure that the SDK shuts down cleanly and has a chance to deliver analytics events
+  // to LaunchDarkly before the program exits. If analytics events are not delivered, the user
+  // properties and flag usage statistics will not appear on your dashboard. In a normal
+  // long-running application, the SDK would continue running and events would be delivered
+  // automatically in the background.
+
+  await LDClient.close();
```
Comment (scribblingfox) [inline lib/main.dart:59]: Just stuck this at the end, but could it possibly go in somewhere after line 49?

File: README.md [hunk 4]
```diff
@@ -1,17 +1,13 @@
-# LaunchDarkly Sample Flutter Application
+# LaunchDarkly sample Flutter application
 
 We've built a simple Flutter application that demonstrates how LaunchDarkly's SDK works.
 
-Below, you'll find the basic build procedure, but for more comprehensive instructions, you can visit the [Flutter SDK reference guide](https://docs.launchdarkly.com/sdk/client-side/flutter).
+Below, you'll find the basic build procedure. For more comprehensive instructions, you can visit your [Quickstart page](https://app.launchdarkly.com/quickstart#/) or the [Flutter SDK reference guide](https://docs.launchdarkly.com/sdk/client-side/flutter).
```
Comment (mmrj) [inline README.md]:
```suggestion
Below, you'll find the build procedure. For more comprehensive instructions, you can visit your [Quickstart page](https://app.launchdarkly.com/quickstart#/) or the [Flutter SDK reference guide](https://docs.launchdarkly.com/sdk/client-side/flutter).
```
Your review of other variations of this work recommended to not keep both "simple" and "basic," so I've been removing the "basic".

File: README.md [hunk 5]
```diff
@@ -1,17 +1,13 @@
...
 
 ## Build instructions
 
-1. Make sure you have [Flutter SDK](https://flutter.dev/docs/get-started/install) installed.
+1. Edit `lib/main.dart` and set the value of `MOBILE_SDK_KEY` to your LaunchDarkly mobile key. If there is an existing boolean feature flag in your LaunchDarkly project that you want to evaluate, set `FEATURE_FLAG_KEY` to the flag key.
 
-2. Edit `lib/main.dart` and set the value of `MOBILE_SDK_KEY` to your LaunchDarkly mobile key. If there is an existing boolean feature flag in your LaunchDarkly project that you want to evaluate, set `FEATURE_FLAG_KEY` to the flag key.
+2. On the command line, run `flutter run`.
```
Comment (mmrj) [inline README.md:11]: Does this start up anything for you? (I didn't try locally.) I'm wondering if we should keep the existing step 3 that tells folks to open a simulator instance.

File: lib/main.dart (hunk 2 above)
Comment (mmrj) [reply lib/main.dart:44]: I'm also not sure whether it's typical to call `setState()` twice in a row like this? vs. just calling once and displaying two messages in the callback (assuming this is the right place to show a message on initialization failure -- like @scribblingfox , I'm not sure on that).

**Review by mmrj — COMMENTED** [review]
Minor comments on the readme. I don't have enough background here to answer your questions in `main.dart`.

File: lib/main.dart (hunk 2 above)
Comment (scribblingfox) [reply lib/main.dart:44]: I had that thought too! I wasn't sure how to initiate a message appended to another, though.

**Comment by kinyoklion** [comment]
This repo is now deprecated.

**Pull request closed without merging**